### PR TITLE
cmake: fix non-mpi build

### LINKED ISSUE #193: Long Range Example Requires MPI even when MPI is disabled
The spme example depends on particles.h, which includes mpi. It's not conditional on MPI like the `directsum_example`

https://github.com/ECP-copa/Cabana/blob/500326520268bcaa05e77ca8c9a6a7cb7320230b/core/example/longrange/CMakeLists.txt#L10

### PR DESCRIPTION
**Pull request by junghans**
Fix #193

### CODE REVIEW

File: core/example/longrange/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,11 @@
-if(Cabana_ENABLE_MPI)
-  add_executable(Direct directsum_example.cpp)
-  target_link_libraries(Direct cabanacore)
-  add_executable(Ewald ewaldsum_example.cpp)
-  target_link_libraries(Ewald cabanacore)
+if(NOT Cabana_ENABLE_MPI)
+  return()
```
Comment (dalg24) [inline core/example/longrange/CMakeLists.txt]: Come on :@

Comment (dalg24) [reply core/example/longrange/CMakeLists.txt]: Don't include the subdirectory then

Comment (junghans) [reply core/example/longrange/CMakeLists.txt]: :-)

Comment (junghans) [reply core/example/longrange/CMakeLists.txt]: We do the same in a couple of places:
```
git grep 'return()' -- '*CMakeLists.txt'
core/example/CMakeLists.txt:  return()
core/fortran-example/tutorial/05_parallel_for/CMakeLists.txt:    return()
core/fortran-example/tutorial/06_kokkos_parallel_for/CMakeLists.txt:  return()
core/performance_test/peakflops/CMakeLists.txt:  return()
core/performance_test/peakflops/CMakeLists.txt:    return()
core/unit_test/CMakeLists.txt:  return()
```

Comment (junghans) [reply core/example/longrange/CMakeLists.txt]: I prefer this way as it localizes the restriction more, but I am happy to exclude the subdir.

Comment (dalg24) [reply core/example/longrange/CMakeLists.txt]: Fine

**Pull request merged**